File: JavaScript/SavingsAcct.js
import InterestRewardsAcct from "./InterestRewardsAcct.js";

export default class SavingsAcct extends InterestRewardsAcct {
    constructor(initialAmount, acctName) {
        super(initialAmount, acctName);
        this.fee = 5;
    }

    withdraw(amount) {
        try {
            this.viableTransaction(amount + this.fee);
            this.balance = this.balance - (amount + this.fee);
            console.log("\nWithdraw completed.");
            this.getBalance();
        } catch (err) {
            console.log("\nWithdraw interrupted: " + err);
        }
    }
}

File: Java/SavingsAcct.java
public class SavingsAcct extends InterestRewardsAcct {
    double fee;

    SavingsAcct(double initialAmount, String acctName) {
        super(initialAmount, acctName);
        fee = 5;
    }

    @Override
    public void withdraw(double amount) {
        try {
            this.viableTransaction(amount + fee);
            balance = balance - (amount + fee);
            System.out.println("\nWithdraw completed.");
            this.getBalance();
        } catch (BalanceException e) {
            System.out.println("\nWithdraw interrupted: " + e);
        }
    }
}
